doc/nrf/ug_pelion.rst

```diff
@@ -0,0 +1,104 @@
+********
+
+Pelion is an IoT cloud platform that offers services in the fields of connectivity, device management, and edge computing, which allow you to remotely manage and update your IoT devices.
+To read more, see the `Pelion website`_ and the `Pelion Device Management documentation`_.
+
+An example of the application that is using the Pelion Device Management library in |NCS| is :ref:`pelion_client`.
+
+Integration with |NCS|
+======================
+
+Pelion Device Management library (``mbed-cloud-client`` library) is included in the |NCS| as one of the submodule repositories managed with the :ref:`zephyr:west` tool.
+It is fetched into the :file:`ncs/modules/lib/pelion-dm` directory of the |NCS|.
+
+Later on this page, this library is referred to as *Pelion Device Management library*.
+
+Pelion Device Management requirements
+=====================================
+
+To use Pelion Device Management library you need a developer account on the `Pelion Device Management Portal`_.
+
+Configuration
+*************
+
+The Pelion Device Management library is enabled with :option:`CONFIG_PELION_CLIENT` configuration option.
+
+For the library to work, you must enable and properly configure the Mbed TLS library.
+To connect to the Pelion Device Management Portal, the device must be provisioned with valid credentials.
+For the firmware update procedure to work, you must provision the device with valid update resources.
+
+The Pelion Device Management library needs a dedicated partition on non-volatile storage.
+The library also requires correct configuration of heap.
+
+Finally you need to enable and properly configure the network backend.
+
+Pelion Device Management library configuration
+==============================================
+
+The Pelion Device Management library features can be controled using Kconfig options.
+These options are defined within the Pelion Device Management library repository.
+
+Below is the list of some of the important ones:
+* :option:`CONFIG_PELION_UPDATE` - enables device firmware update (DFU) feature.
+* :option:`CONFIG_PELION_TRANSPORT_MODE_TCP`, :option:`CONFIG_PELION_TRANSPORT_MODE_UDP`, :option:`CONFIG_PELION_TRANSPORT_MODE_UDP_QUEUE`- choice selecting the transport protocol used by the library.
+
+For more information and a full list of the library configuration options, read the `Zephyr integration tutorial`_ in the Pelion documentation.
+
+Mbed TLS
+========
+
+The Pelion Device Management library is depending on a properly configured Mbed TLS library.
+To simplify the development within the |NCS|, a predefined set of Mbed TLS configuration options was prepared.
+Set compatible with :ref:`nrfxlib:nrf_security` can be enabled using :option:`CONFIG_PELION_NRF_SECURITY` configuration option.
+
+Make sure to properly configure memory region used by Mbed TLS library for a dynamic allocations.
+Memory for this library is allocated from either the libc heap or a dedicated buffer, depending on a configuration.
```

Comment:
As discussed f2f: documentation page refs for these options are missing.
Can we mention the Kconfigs for the configuration options?

Reply:
expanded somehow